### PR TITLE
Domains: Fix app hanging when selecting domain by showing an error notice

### LINKED ISSUE #20748: App hangs when buying a domain
### Expected behavior

After choosing a domain e.g. hometitle.org, the app should advance to the checkout screen to allow me to complete the purchase.

### Actual behavior

The app hangs with a spinner if I choose one of the domains that appears by default on the "Search domains" screen.

### Steps to reproduce the behavior
1. Tap the **Find a custom domain** dashboard card
2. Tap **Search for a domain**
3. Tap the first domain suggestion
4. Tap **Select domain**
5. Notice the spinner is shown indefinitely and the app doesn't progress to the next screen

### Additional info

WP.com says "Sorry, the domain you are trying to add can not be bought on WordPress.com at this time." when trying to buy this domain. While WP.com doesn't suggest this domain by default like mobile does, users could come across this domain (or other unavailable domains), so we could handle this case gracefully on mobile by showing the same message.

Screenshot 2023-05-26 at 17 04 23

Related Android issue: https://github.com/wordpress-mobile/WordPress-Android/issues/18538

### PR DESCRIPTION
**Pull request by staskus**
Fixes #20748

To test:

After choosing a domain e.g. hometitle.org, the app should advance to the checkout screen to allow me to complete the purchase.

1. Tap the Find a custom domain dashboard card
2. Tap Search for a domain
3. Tap the first domain suggestion
4. Tap Select domain
5. Confirm that error notice appears and spinner disappears

## Regression Notes
1. Potential unintended areas of impact

None, we're adding additional handling to before unhandled code path

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/bb271df9-85de-495f-a497-36e827b0dab6

image

### CODE REVIEW

**Comment by guarani** [comment]
Thanks for this, @staskus! I didn't get to this today but will review it ASAP.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 22.6
 -----
 * [**] [internal] Check required WordPress version to set "galleryWithImageBlocks" flag [#20736]
-
+* [**] [internal] [Jetpack-only] Domains selection: Show error message when selecting unsupported domains. [#20786]
```
Comment (guarani) [inline RELEASE-NOTES.txt]: nitpick: I think the `[internal]` is for stuff we want to keep out of the public release notes (see pdcxQM-Km-p2). That doesn't seem to be the case here, although changing it to `*` would lower its priority in the public release notes, which might be what we want. Wdyt?

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift [hunk 2]
```diff
@@ -360,6 +377,13 @@ extension RegisterDomainSuggestionsViewController {
         static let primaryButtonTitle = NSLocalizedString("Select domain",
                                                           comment: "Register domain - Title for the Choose domain button of Suggested domains screen")
         static let supportButtonTitle = NSLocalizedString("Help", comment: "Help button")
+
+        static let errorTitle = NSLocalizedString("domains.failure.title",
+                                                  value: "Sorry, the domain you are trying to add can not be bought on the Jetpack app at this time.",
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift]: nitpick: "cannot" is [the more common spelling](https://www.thesaurus.com/e/grammar/cannot-or-can-not/) of "can not", although this is how WP.com put it.

**Review by guarani — APPROVED** [review]
Works great! I left a couple of nitpicks so feel free to merge as-is or apply them before merging.

Thanks @staskus!

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (staskus) [reply RELEASE-NOTES.txt]: Thanks for sharing the resource! Yes, you're right, I'll update the notes!

File: WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift (hunk 2 above)
Comment (staskus) [reply WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainSuggestionsViewController.swift]: Yes, it looked strange to me too but I didn't want to dispute the WP.com translations. However, we can update it in the app to have a more common spelling.

**Pull request merged**